include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -19,6 +19,15 @@ namespace nbl::video
 struct NBL_API2 ICleanup
 {
     virtual ~ICleanup() = 0;
+
+    std::unique_ptr<ICleanup> next;
+
+    static void chain(std::unique_ptr<ICleanup>& first, std::unique_ptr<ICleanup>&& next)
+    {
+        if (first)
+            return chain(first->next, std::move(next));
+        first = std::move(next);
+    }
```

Comment:
As already talked about, this is extremely sketchy:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/517#discussion_r1264588641

also the reason you have:
- a non const `m_cachedCreationParams`
- a public `chainPreDestroyCleanup` method

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/517#discussion_r1264588772

Reply:
Another fun comment: https://github.com/Devsh-Graphics-Programming/Nabla/pull/517#discussion_r1264665853

My guess is that the ICleanup which ties CUDA memory to an IDeviceMemoryBacked needs to go like this:
```
MemoryBacked -regular smartpointer reference-> Imported MemoryAllocation -Cleanup-> Shared CUDA Memory
```

so this cleanup chain and functions being in IDeviceMemoryBacked is a big mistake... its the imported memory that should have a post-destroy cleanup in the creation params, nothing else

Reply:
w.r.t CloseHandle, do you need to call it 1 time after all users are done (original creator and importers), or N times per N users?

Does the need to call CloseHandle arise from creating an exportable Vulkan memory, or after querying the handle?

etc. etc.